src/src/tx_main.cpp

```diff
@@ -384,11 +385,44 @@ void ICACHE_RAM_ATTR HandlePrepareForTLM()
   // If next packet is going to be telemetry, start listening to have a large receive window (time-wise)
   if (ExpressLRS_currAirRate_Modparams->TLMinterval != TLM_RATIO_NO_TLM && modresult == 0)
   {
+    Radio.ClearIrqStatus(SX1280_IRQ_RADIO_ALL);
+    Radio.SetDioIrqParams(SX1280_IRQ_RX_DONE, SX1280_IRQ_RX_DONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE);
     Radio.RXnb();
     TelemetryRcvPhase = ttrpInReceiveMode;
   }
 }
 
+int8_t ICACHE_RAM_ATTR PowerEnumToLBTLimit(PowerLevels_e txPower)
+{
+    // Calculated from EN 300 328, assuming 800kHz BW for sx1280
+    // TL = -70 dBm/MHz + 10 × log10 (100 mW / Pout) (Pout in mW e.i.r.p.)
+    // Values above 100mW are not relevant, default to 100mW threshold
+    // TODO: This threshold should be modified with a config for antenna gain
+    switch(txPower)
+    {
+    case PWR_10mW: return -60;
+    case PWR_25mW: return -64;
+    case PWR_50mW: return -67;
+    case PWR_100mW: return -70;
+    default: return -70;
+    }
+}
+
+volatile uint32_t RxStartTimeCCA = 0;
+
+void ICACHE_RAM_ATTR BeginClearChannelAssessment()
+{
+  // Listen Before Talk (LBT) aka clear channel assessment (CCA)
+  // Not interested in packets or interrupts while measuring RF energy on channel.
+  Radio.SetDioIrqParams(SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE);
+  Radio.RXnb();
+}
+
+bool ICACHE_RAM_ATTR ChannelIsClear()
+{ 
+  return Radio.GetRssiInst() < PowerEnumToLBTLimit((PowerLevels_e)POWERMGNT::currPower());
```

Comment:
Isnt there a sampling duration requirement? 18us?

Reply:
yes
> The CCA observation time shall be not less than 0,2 % of the Channel Occupancy Time with a minimum of 18 μs

per my calculations, since the sx1280 spends minimum one lora symbol-time to measure rssi, we should be OK for all modes as long as we have waited long enough to get one valid rssi measurement. The rssi measurements seems to be an average value over one lora symbol-time. which should be ok to use as LBT sensing. See calculations below:

our channel occupancy time and the 0.2% minimum listening time for the different modes are:

- 1665 * 4 = 6660 ->  18us is minimum
- 3300 * 4 = 13200 -> 26.4us
- 5871 * 4 = 18443 -> 37us
- 18443 * 2 = 36886 -> 73.8us

symbol time for same modes:
- 39.4us
- 78.8us
- 157.6us
- 630.5us

These are always greater than minimum listening time, so it should be OK?

Reply:
![image](https://user-images.githubusercontent.com/14170229/148456122-dc879a08-078f-4dd3-ad74-1fbbe7b6ac7d.png)

Reply:
Thanks.  Lets leave this comment unresolved just so your explanation is clearly visible to others.